static/app/components/modals/metricWidgetViewerModal.tsx

```diff
@@ -56,13 +61,15 @@ function MetricWidgetViewerModal({
   const [metricWidgetQueryParams, setMetricWidgetQueryParams] =
     useState<MetricWidgetQueryParams>(convertToMetricWidget(widget));
 
-  const defaultTitle = useMemo(
-    () => stringifyMetricWidget(metricWidgetQueryParams),
+  const widgetMQL = useMemo(
+    () => getFormattedMQL(metricWidgetQueryParams),
     [metricWidgetQueryParams]
   );
 
-  const [title, setTitle] = useState<string>(widget.title ?? defaultTitle);
-  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const [editedTitle, setEditedTitle] = useState<string>(widget.title || widgetMQL);
+  const isMQLTitle = useMemo(() => isFormattedMQL(editedTitle), [editedTitle]);
+  // If user renamed the widget, dislay that title, otherwise display the MQL
+  const titleToDisplay = isMQLTitle ? widgetMQL : editedTitle;
```

Comment:
not sure if I agree with this logic 🤔 This means, if they only remove e.g the query part of the mql we will not save / display it, right?
What was the problem with the previous logic?

Reply:
To simplify I would suggest
```suggestion
  const titleToDisplay = editedTitle === '' ? widgetMQL : editedTitle;
```

Reply:
Correct, but Well with the previous logic if the user had a widget with a MQL title that was saved and then they later decided to edit it by adding lets say group by, that will not be persisted in the title because `newMQL !== oldMQL`. But tbh i am not sure which one is better 🤷

Reply:
hmm imho, ideally we only save the edited title once they enter one (not sure about the current behaviour here).
Once there is a custom title we should never auto-update it.
🤔

Reply:
Haven't thought of that, so far we always save a title, whether auto-generated or not, but let me edit that! Thanks!